### PR TITLE
Updates README.md, CONTRIBUTING.md and BUILDING.md

### PR DESCRIPTION
**Pull request by pivotal-jbarrett**
* Updates established versioning rules.
* Fixes links to CI badge.
* Fixes links to internal anchors.
* Fixes links to logo.
* Updates OS versions.
* Updates tooling versions.
* Moves contributing details from BUILDING.md to CONTRIBUTING.md.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,27 +1,53 @@
...
+cover [ABI](https://en.wikipedia.org/wiki/Application_binary_interface) compatibility.
+
+## C++
+
+ABI compatibility is not preserved similarly to the API compatibility. ABI compatibility is only guaranteed between
+patch releases. Recompilation is required for both minor and major releases.
+
+## .NET
+
+ABI compatibility is preserved similarly to the API compatibility. ABI compatibility is guaranteed between path and
```
Comment (pdxcodemonkey) [inline README.md:47]: Typo here, I think - should be `patch`, not `path`.

**Review by pdxcodemonkey — CHANGES_REQUESTED** [review]
One typo to fix, rest looks good.  Nice updates to instructions, thanks!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -156,7 +217,8 @@ When writing new or refactoring old code please make the following changes.
     }
     ```
 
-  * Fix [`std::string::c_str()`](http://en.cppreference.com/w/cpp/string/basic_string/c_str) calls where passed to [`std::string`](http://en.cppreference.com/w/cpp/string/basic_string) parameters.
+  * Fix [`std::string::c_str()`](http://en.cppreference.com/w/cpp/string/basic_string/c_str) calls where passed to
```
Comment (davebarnes97) [inline CONTRIBUTING.md:220]: When I view this section using macdown, this bullet and those that follow are indented under the preceding text. I'm not sure if this is what you intended. LMK if you'd like me to reformat this section.

Screen Shot 2021-03-02 at 9 53 18 AM

Comment (moleske) [reply CONTRIBUTING.md:220]: If you use the "Rich Diff" button in the the GitHub file view, the indentation doesn't occur.  I think you're seeing a difference with between GitHub and macdown markdown

Comment (davebarnes97) [reply CONTRIBUTING.md:220]: I see what you mean -- macdown wants a lot more whitespace.
OK, close enough for rock'n'roll. I'll drop this concern.

**Review by davebarnes97 — APPROVED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (pivotal-jbarrett) [reply CONTRIBUTING.md:220]: For some reason there was a single space before `*` on a few lines then two spaces on the others. I have corrected so we will see if they display consistently now.

**Review by pdxcodemonkey — APPROVED** [review]

**Pull request merged**